org.osgi.service.jakartaws/src/org/osgi/service/jakartaws/whiteboard/SoapWhiteboardConstants.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0 
+ *******************************************************************************/
+package org.osgi.service.jakartaws.whiteboard;
+
+/**
+ * Defines standard constants for the Jakarta Web Services Whiteboard services.
+ * 
+ * @author $Id$
+ */
+public class SoapWhiteboardConstants {
```

Comment:
Is it only SOAP that is supported? Isn't the jakarta specification technically independent of SOAP, and allows XML/HTTP binding without SOAP?

Reply:
SOAP is an envelope format based on XML. Webservices defines the Exchange of SOAP Messages via a given transport (that can be HTTP or JMS or ...) the [specification website](https://projects.eclipse.org/projects/ee4j.jaxws) says:

> The Jakarta XML Web Services is a Java programming language API for creating web services, particularly SOAP services.

Reply:
If you are interested in the full details this gives a good overview:

https://jakarta.ee/specifications/xml-web-services/4.0/jakarta-xml-ws-spec-4.0.html#introduction

basically one could also use XML-RPC its been a while since I've seen any (working) implementation for modern java, the official RI seems this one and jax-ws only reference that as an alternative binding https://github.com/eclipse-ee4j/jax-rpc-ri

maybe because most of the time SOAP (that supports Message and RPC style) is mentioned as the successor of XML-RPC. At least this specification stems from an implementation that based on SOAP and therefore has this naming.

Reply:
I have now renamed all SOAP > Webservice